### ISSUE TITLE
why is the code not same as the paper

### ISSUE THREAD

**Issue by MrCrazyCrab** [dlinzhao/JSNet#9]
in the paper, you said the sem would get through  1D Convolution with non-linear operation to the ins,  and as it did in the code. However, the output of the net work has size of  B*n*13 and the ins have the size of b*n*5. i wonder Why is the output the opposite between ins and sem. i always thought the sem means the 13.

**Comment by dlinzhao** [comment]
@MrCrazyCrab The instance features and semantic features have the same dimension 128 (e.g. 24 x 4096 x 128). At last, we apply a 1D convolution to the instance features for generating the instance embeddings (B x N x 128 --> B x N x 5).  Similarly, we obtain the semantic feature matrices (B x N x 128 --> B x N x 13) by a 1D convolution. The description in the paper is consistent with the code.

**Comment by MrCrazyCrab** [comment]
@dlinzhao i konw what's you meaning, just a little confuse. Pardon for my ignorance， i thing your idea is the same as ASIS, but i met some problem in your code. When i train the model, i found the the label of seg is only 0 or 1 after 1 epoch, which would not happen in ASIS. I I have tested the two algorithms using the same data,  so is there anything wrong?
looking for your reply!

**Comment by dlinzhao** [comment]
@MrCrazyCrab Can you provide your dataset organized structure, training command and the training log information?

**Comment by MrCrazyCrab** [comment]
@dlinzhao  the structure of my dataset is same sa S3DIS,  i just use some of data , the logs is as follows:
Namespace(batch_size=3, decay_rate=0.7, decay_step=200000, gpu=0, input_list='ins_val.txt', learning_rate=0.001, log_dir='logs', max_epoch=1000, momentum=0.9, num_point=4096, optimizer='adam', restore_model='log_01')
Fail to load pre_modelfile: log_01
**** EPOCH 000 ****
---------------------------------
2020-11-16 16:03:30.554125
loss: nan; sem_loss: 3.681; disc_loss: 28.025; correct: 0.029785; total_IoU:0.015118.
loss: nan; sem_loss: 1.750; disc_loss: 9.000; correct: 0.395181; total_IoU:0.246247.
loss: nan; sem_loss: 0.826; disc_loss: 9.000; correct: 0.361427; total_IoU:0.220574.
accuracy: 0.356666
mean loss: nan
one epoch time: 39.8148
**** EPOCH 001 ****
---------------------------------
2020-11-16 16:04:10.368989
loss: nan; sem_loss: 0.808; disc_loss: 9.000; correct: 0.307454; total_IoU:0.181652.
loss: nan; sem_loss: 0.744; disc_loss: 9.000; correct: 0.293260; total_IoU:0.171825.
loss: nan; sem_loss: 0.728; disc_loss: 9.000; correct: 0.301841; total_IoU:0.177746.
accuracy: 0.303432
mean loss: nan
one epoch time: 31.9456
**** EPOCH 002 ****
---------------------------------
2020-11-16 16:04:42.314727
loss: nan; sem_loss: 0.720; disc_loss: 9.000; correct: 0.327555; total_IoU:0.195854.
loss: nan; sem_loss: 0.713; disc_loss: 9.000; correct: 0.282976; total_IoU:0.164806.
loss: nan; sem_loss: 0.704; disc_loss: 9.000; correct: 0.320766; total_IoU:0.191019.
accuracy: 0.320853
mean loss: nan
one epoch time: 32.0504
**** EPOCH 003 ****
---------------------------------
2020-11-16 16:05:14.365208
loss: nan; sem_loss: 0.705; disc_loss: 9.000; correct: 0.323405; total_IoU:0.192894.
loss: nan; sem_loss: 0.703; disc_loss: 9.000; correct: 0.333071; total_IoU:0.199811.
loss: nan; sem_loss: 0.705; disc_loss: 9.000; correct: 0.326011; total_IoU:0.194751.
accuracy: 0.326809
mean loss: nan
one epoch time: 31.7342
![微信图片_20201116160825](https://user-images.githubusercontent.com/52489106/99227281-fdaff400-2825-11eb-885d-8fd1479686c3.png)

**Comment by dlinzhao** [comment]
@MrCrazyCrab It seems that you only use the Network of JSNet and use your own training pipeline. From the output information, the sem_loss and disc_loss are correct, but the total loss is Nan. So you can check the calculation of total Loss or use the training pipeline in JSNet.

**Comment by MrCrazyCrab** [comment]
i don't think the sem_loss and disc_loss are correct. The output of train phase label is invalid. Is it only in the later epoch that the network would get label right?

**Comment by dlinzhao** [comment]
I mean there is no numerical error (e.g. Nan or Inf) for the sem_loss and disc_loss. The total loss is Nan causing the network to fail to output correct predicting labels.

**Comment by MrCrazyCrab** [comment]
@dlinzhao  for the loss, i found that the nan is comes from myself designed loss function. And i comment my loss function, the logs is as follows:
Namespace(batch_size=3, decay_rate=0.7, decay_step=200000, gpu=0, input_list='ins_val.txt', learning_rate=0.001, log_dir='logs', max_epoch=1000, momentum=0.9, num_point=4096, optimizer='adam', restore_model='log_01')
Fail to load pre_modelfile: log_01
**** EPOCH 000 ****
---------------------------------
2020-11-16 17:13:30.147359
loss: 34.298; sem_loss: 3.457; disc_loss: 30.840; correct: 0.037109; total_IoU:0.018905.
loss: 10.657; sem_loss: 1.663; disc_loss: 8.994; correct: 0.094529; total_IoU:0.049609.
loss: 9.801; sem_loss: 0.809; disc_loss: 8.991; correct: 0.112574; total_IoU:0.059644.
accuracy: 0.117682
mean loss: 10.984212
one epoch time: 32.3599
**** EPOCH 001 ****
---------------------------------
2020-11-16 17:14:02.507317
loss: 9.772; sem_loss: 0.793; disc_loss: 8.979; correct: 0.245036; total_IoU:0.139624.

 but the label is still have two, the pred is invalid.

**Comment by dlinzhao** [comment]
At least the current total loss is normal (not Nan and Inf). You can test it after more training epoch.
